src/BaseActions.tsx

```diff
@@ -42,3 +46,20 @@ export function replace(name: string, params?: object): Action {
 export function reset(state: PartialState & { key?: string }): Action {
   return { type: 'RESET', payload: state };
 }
+
+export function setParams(
+  params: object,
+  target: { name?: string; key?: string }
```

Comment:
I think we should make the signature same as `navigate` etc. I think it's fine if it's required.